server/data/athenaClient.ts

```diff
@@ -98,10 +98,11 @@ export default class AthenaClient {
       // eslint-disable-next-line no-await-in-loop
       const response = await this.athena.send(command)
       nextToken = response.NextToken
-      const rows = response.ResultSet.Rows || []
+      const rows = response.ResultSet?.Rows || []
       // eslint-disable-next-line no-restricted-syntax
       for (const row of rows) {
-        yield row.Data.map(datum => datum.VarCharValue)
+        const data = row.Data || []
+        yield data.map(datum => datum.VarCharValue)
```

Comment:
if you only `yield` if row.Data is present do you still need to add `|void` to the return type? e.g. only yield `string` values if present and skip the rest.

Reply:
`AsyncGenerator<string[], void, void>` means it yields an array of strings, returns void (there is no return statement) and takes void parameters if you call `.next()` on the async generator.

Reply:
also i'm cheating again. `datum.VarCharValue` can be undefined according to the the AWS lib. so really, the return type should be `AsyncGenerator<(string | undefined)[], void, void>` (but it'd be more annoying to use and i am trusting AWS to return correct values!)

Reply:
> `AsyncGenerator<string[], void, void>` means it yields an array of strings, returns void (there is no return statement) and takes void parameters if you call `.next()` on the async generator (using a for loop passes void into this `.next` method).

This is VERY interesting, I'm aware of the fact that you can return from an async generator but I assumed it was the same type as the values you yield - I guess that's what happen most of the time (IF you return) but it makes this more flexible.

Also, I completely removed the fact that you can pass values IN a generator by passing something to `next()` - that always feels weird.